Algorithm/Searching_Sorting/fibonacci_search.cpp

```diff
@@ -0,0 +1,161 @@
+when we have to search the element on the smaller section of the array and hence we get our average case complexity as O(log n).
+
+According to the algorithm we will first sort the array.
+Output is based on Sorted array.
+
+QUESTION-->
+1) You are given 't' test cases to check.
+2) You are given a memory space of array size, 10.
+3) You are asked to enter elements in the array.
+4) You can input as much elements in the array less than or equal to 10.
+5) The array should be sorted if Unsorted.
+6) You have to chose an element and find its position in the sorted array.
+7) if element not found, print -1.
+    */
+
+#include <bits/stdc++.h>
+using namespace std;
+
+// function created to find the min value between x and y
+int min(int x, int y) { return (x <= y) ? x : y; }
+
+//function created returns index of x if present,  else returns -1.
+int FibonacciSearch(int arr[], int x, int n)
+{
+    // a,b,c are variables that stores the fibonacci numbers sequentially.
+    int a = 0;
+    int b = 1;
+    int c = a + b;
+
+    // until c does not become equal to or greater than n , loop executes.
+    while (c < n)
+    {
+
+        a = b;
+        b = c;
+        c = a + b;
+    }
+    // Marks the eliminated range from front.
+    int offset = -1;
+
+    // checking if c is at valid location.
+    while (c > 1)
+    {
+        // i will be assigned the value of min() used.
+        int i = min(offset + a, n - 1);
+
+        /*If x is greater than the value at index ,c cut the subarray array from offset to i*/
+        if (arr[i] < x)
+        {
+
+            c = b;
+            b = a;
+            a = c - b;
+            offset = i;
+        }
+        /*If x is greater than the value at index ,c
+cut the subarray after i+1. */
+        else if (arr[i] > x)
+        {
+            c = a;
+            b = b - a;
+            a = c - b;
+        }
+
+        else
+            //if element found, return index.
+            return i;
+    }
+    //comparing the last element with x
+    if (b && arr[offset + 1] == x)
+        return offset + 1;
+    // if not then return -1.
+    return -1;
+}
+
+int main()
+{
+    int t;
+    cin >> t;
+    while (t--)
+    {
+        int l;
+        cout << "\nEnter the number of elements in array which should be less than 10";
+        cin >> l;
+        //array defined
+        int arr[10];
+        cout << "Enter elements in array";
+        for (int i = 0; i < l; i++)
+        {
+            cin >> arr[i];
+        }
+        //sorting the array
+        sort(arr, arr + l);
+
+        int n = sizeof(arr) / sizeof(arr[0]);
+        int x;
+        cout << "\nEnter element to be searched :";
+        cin >> x;
+        cout << "Found at index:" << FibonacciSearch(arr, x, n);
+    }
+    return 0;
+}
+
+/* complexities */
+
+/* Worst case time complexity:  O(log n)
+Average case time complexity:  O(log n)
+Best case time complexity:  O(1)
+Space complexity:  O(1)   */
+
+/* sample input :

```

Comment:
Follow 
Test Cases:
Input 
Output

Include all the part below code in single multiline comment